### ISSUE TITLE
Abort for high column number when running dtest limits.py:TestLimits.max_columns_and_query_parameters_test

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#916]
```
(gdb) bt
#0  0x0000000000757ca0 in managed_bytes::free_chain (p=<optimized out>, this=<optimized out>) at ./utils/managed_bytes.hh:117
#1  0x00000000007677a6 in ~managed_bytes (this=<optimized out>, __in_chrg=<optimized out>) at ./utils/managed_bytes.hh:195
#2  ~atomic_cell_or_collection (this=<optimized out>, __in_chrg=<optimized out>) at atomic_cell_or_collection.hh:30
#3  pop_back (this=0x600014aa5458, this=0x600014aa5458) at utils/managed_vector.hh:213
#4  clear (this=this@entry=0x600014aa5458) at utils/managed_vector.hh:170
#5  managed_vector<atomic_cell_or_collection, 5u, unsigned int>::clear_and_release (this=this@entry=0x600014aa5458)
    at utils/managed_vector.hh:77
#6  0x0000000000759b01 in ~managed_vector (this=0x600014aa5458, __in_chrg=<optimized out>) at utils/managed_vector.hh:127
#7  row::~row (this=0x600014aa5450, __in_chrg=<optimized out>) at mutation_partition.cc:959
#8  0x000000000076780d in ~deletable_row (this=0x600014aa5430, __in_chrg=<optimized out>) at mutation_partition.hh:383
#9  rows_entry::~rows_entry (this=0x600014aa5400, __in_chrg=<optimized out>) at mutation_partition.hh:477
#10 0x00000000007678ac in destroy<rows_entry> (obj=0x600014aa5400, this=0x19a3860 <standard_allocation_strategy_instance>)
    at ./utils/allocation_strategy.hh:118
#11 operator() (obj=0x600014aa5400, __closure=<optimized out>) at ./utils/allocation_strategy.hh:166
#12 operator() (p=<synthetic pointer>, this=<synthetic pointer>) at /usr/include/boost/intrusive/detail/node_cloner_disposer.hpp:100
#13 dispose_subtree<boost::intrusive::detail::node_disposer<current_deleter() [with T = rows_entry]::<lambda(rows_entry*)>, boost::intrusive::mhtraits<rows_entry, boost::intrusive::set_member_hook<>, &rows_entry::_link>, (boost::intrusive::algo_types)5u> > (disposer=..., 
    x=0x600014aa5400) at /usr/include/boost/intrusive/bstree_algorithms.hpp:2047
#14 boost::intrusive::bstree_algorithms<boost::intrusive::rbtree_node_traits<void*, false> >::clear_and_dispose<boost::intrusive::detail::node_disposer<auto current_deleter<rows_entry>()::{lambda(rows_entry*)#1}, boost::intrusive::mhtraits<rows_entry, boost::intrusive::set_member_hook<void, void, void, void>, &rows_entry::_link>, (boost::intrusive::algo_types)5> >(boost::intrusive::rbtree_node<void*>* const&, boost::intrusive::detail::node_disposer<auto current_deleter<rows_entry>()::{lambda(rows_entry*)#1}, boost::intrusive::mhtraits<rows_entry, boost::intrusive::set_member_hook<void, void, void, void>, &rows_entry::_link>, (boost::intrusive::algo_types)5>) (header=@0x6000111c49e8: 0x6000123f09f8, 
    disposer=...) at /usr/include/boost/intrusive/bstree_algorithms.hpp:691
#15 0x0000000000759b8b in clear_and_dispose<current_deleter() [with T = rows_entry]::<lambda(rows_entry*)> > (disposer=..., 
    this=0x6000123f09e8) at /usr/include/boost/intrusive/bstree.hpp:1468
#16 mutation_partition::~mutation_partition (this=0x6000123f0970, __in_chrg=<optimized out>) at mutation_partition.cc:110
#17 0x0000000000b13357 in ~data (this=0x6000123f0940, __in_chrg=<optimized out>) at ./mutation.hh:34
#18 std::default_delete<mutation::data>::operator() (__ptr=0x6000123f0940, this=<optimized out>)
    at /usr/include/c++/5.3.1/bits/unique_ptr.h:76
#19 0x0000000000b65266 in ~unique_ptr (this=0x6000112ee188, __in_chrg=<optimized out>) at /usr/include/c++/5.3.1/bits/unique_ptr.h:236
#20 ~mutation (this=0x6000112ee188, __in_chrg=<optimized out>) at ./mutation.hh:32
#21 ~schema_mutations (this=0x6000112ee180, __in_chrg=<optimized out>) at ./schema_mutations.hh:29
#22 ~pair (this=<optimized out>, __in_chrg=<optimized out>) at /usr/include/c++/5.3.1/bits/stl_pair.h:96
#23 destroy<std::pair<db::schema_tables::qualified_name const, schema_mutations> > (this=<optimized out>, __p=<optimized out>)
    at /usr/include/c++/5.3.1/ext/new_allocator.h:124
#24 _S_destroy<std::pair<db::schema_tables::qualified_name const, schema_mutations> > (__p=<optimized out>, __a=...)
    at /usr/include/c++/5.3.1/bits/alloc_traits.h:285
---Type <return> to continue, or q <return> to quit---
#25 destroy<std::pair<db::schema_tables::qualified_name const, schema_mutations> > (__a=..., __p=<optimized out>)
    at /usr/include/c++/5.3.1/bits/alloc_traits.h:414
#26 _M_destroy_node (this=0x6000111c4d20, __p=0x6000112ee140) at /usr/include/c++/5.3.1/bits/stl_tree.h:553
#27 _M_drop_node (__p=0x6000112ee140, this=0x6000111c4d20) at /usr/include/c++/5.3.1/bits/stl_tree.h:561
#28 std::_Rb_tree<db::schema_tables::qualified_name, std::pair<db::schema_tables::qualified_name const, schema_mutations>, std::_Select1st<std::pair<db::schema_tables::qualified_name const, schema_mutations> >, std::less<db::schema_tables::qualified_name>, std::allocator<std::pair<db::schema_tables::qualified_name const, schema_mutations> > >::_M_erase (this=this@entry=0x6000111c4d20, __x=0x6000112ee140)
    at /usr/include/c++/5.3.1/bits/stl_tree.h:1614
#29 0x0000000000b50973 in ~_Rb_tree (this=0x6000111c4d20, __in_chrg=<optimized out>) at /usr/include/c++/5.3.1/bits/stl_tree.h:858
#30 ~map (this=0x6000111c4d20, __in_chrg=<optimized out>) at /usr/include/c++/5.3.1/bits/stl_map.h:96
#31 db::schema_tables::<lambda()>::operator()(void) (__closure=__closure@entry=0x600011090cc0) at db/schema_tables.cc:593
#32 0x0000000000b524d2 in apply (args=<optimized out>, func=<optimized out>) at /home/tgrabiec/src/scylla2/seastar/core/apply.hh:34
#33 apply<db::schema_tables::do_merge_schema(distributed<service::storage_proxy>&, std::vector<mutation>, bool)::<lambda()> > (
    args=<optimized out>, func=<optimized out>) at /home/tgrabiec/src/scylla2/seastar/core/apply.hh:42
#34 do_void_futurize_apply_tuple<db::schema_tables::do_merge_schema(distributed<service::storage_proxy>&, std::vector<mutation>, bool)::<lambda()> > (args=<optimized out>, func=<unknown type in /home/tgrabiec/src/scylla2/build/release/scylla, CU 0x9b45ada, DIE 0x9dbd37c>)
    at /home/tgrabiec/src/scylla2/seastar/core/future.hh:1162
#35 apply<db::schema_tables::do_merge_schema(distributed<service::storage_proxy>&, std::vector<mutation>, bool)::<lambda()> > (
    args=<optimized out>, func=<optimized out>) at /home/tgrabiec/src/scylla2/seastar/core/future.hh:1182
#36 operator() (__closure=<optimized out>) at /home/tgrabiec/src/scylla2/seastar/core/thread.hh:263
#37 std::_Function_handler<void(), seastar::async(Func&&, Args&& ...)::<lambda(seastar::async(Func&&, Args&& ...)::work&)> mutable [with Func = db::schema_tables::do_merge_schema(distributed<service::storage_proxy>&, std::vector<mutation>, bool)::<lambda()>; Args = {}]::<lambda()> >::_M_invoke(const std::_Any_data &) (__functor=...) at /usr/include/c++/5.3.1/functional:1871
#38 0x000000000041627e in operator() (this=0x6000000dfc10) at /usr/include/c++/5.3.1/functional:2271
#39 seastar::thread_context::main (this=0x6000000dfc00) at core/thread.cc:151
#40 0x0000000000517892 in seastar::thread_context::s_main (lo=<optimized out>, hi=<optimized out>) at core/thread.cc:142
#41 0x00007f7f32fa3fa0 in ?? () from /lib64/libc.so.6
#42 0x0000000000000000 in ?? ()

```

**Comment by tgrabiec** [comment]
Another incarnation:

```
(gdb) bt
#0  0x00000000009aee43 in data_value::~data_value (this=0x600013b588d8, __in_chrg=<optimized out>) at types.cc:2633
#1  0x00000000006f504c in ~pair (this=0x600013b588c8, __in_chrg=<optimized out>) at /usr/include/c++/5.3.1/bits/stl_pair.h:96
#2  destroy<std::pair<basic_sstring<char, unsigned int, 15u> const, data_value> > (this=<synthetic pointer>, __p=<optimized out>)
    at /usr/include/c++/5.3.1/ext/new_allocator.h:124
#3  _S_destroy<std::pair<basic_sstring<char, unsigned int, 15u> const, data_value> > (__p=<optimized out>, __a=<synthetic pointer>)
    at /usr/include/c++/5.3.1/bits/alloc_traits.h:285
#4  destroy<std::pair<basic_sstring<char, unsigned int, 15u> const, data_value> > (__a=<synthetic pointer>, __p=<optimized out>)
    at /usr/include/c++/5.3.1/bits/alloc_traits.h:414
#5  _M_deallocate_node (this=0x600015d24008, __n=0x600013b588c0) at /usr/include/c++/5.3.1/bits/hashtable_policy.h:1973
#6  _M_deallocate_nodes (this=0x600015d24008, __n=<optimized out>) at /usr/include/c++/5.3.1/bits/hashtable_policy.h:1986
#7  clear (this=0x600015d24008) at /usr/include/c++/5.3.1/bits/hashtable.h:1914
#8  ~_Hashtable (this=0x600015d24008, __in_chrg=<optimized out>) at /usr/include/c++/5.3.1/bits/hashtable.h:1225
#9  ~unordered_map (this=0x600015d24008, __in_chrg=<optimized out>) at /usr/include/c++/5.3.1/bits/unordered_map.h:98
#10 ~result_set_row (this=0x600015d24000, __in_chrg=<optimized out>) at ./query-result-set.hh:47
#11 _Destroy<query::result_set_row> (__pointer=<optimized out>) at /usr/include/c++/5.3.1/bits/stl_construct.h:93
#12 __destroy<query::result_set_row*> (__last=<optimized out>, __first=0x600015d24000) at /usr/include/c++/5.3.1/bits/stl_construct.h:103
#13 _Destroy<query::result_set_row*> (__last=<optimized out>, __first=<optimized out>) at /usr/include/c++/5.3.1/bits/stl_construct.h:126
#14 _Destroy<query::result_set_row*, query::result_set_row> (__last=0x600015f23fc0, __first=<optimized out>) at /usr/include/c++/5.3.1/bits/stl_construct.h:151
#15 ~vector (this=0x600013b793c8, __in_chrg=<optimized out>) at /usr/include/c++/5.3.1/bits/stl_vector.h:424
#16 query::result_set::~result_set (this=0x600013b793c0, __in_chrg=<optimized out>) at ./query-result-set.hh:101
#17 0x0000000000b44828 in db::schema_tables::create_table_from_mutations (sm=..., version=std::experimental::optional<utils::UUID> [no contained value])
    at /home/tgrabiec/src/scylla2/seastar/core/sstring.hh:240
#18 0x0000000000b456d0 in db::schema_tables::merge_tables(distributed &, <unknown type in /home/tgrabiec/src/scylla2/build/release/scylla, CU 0x9b4600a, DIE 0x9ed63d6>, <unknown type in /home/tgrabiec/src/scylla2/build/release/scylla, CU 0x9b4600a, DIE 0x9ed63db>) (proxy=..., 
    before=before@entry=<unknown type in /home/tgrabiec/src/scylla2/build/release/scylla, CU 0x9b4600a, DIE 0x9ed63d6>, 
    after=after@entry=<unknown type in /home/tgrabiec/src/scylla2/build/release/scylla, CU 0x9b4600a, DIE 0x9ed63db>) at db/schema_tables.cc:692
#19 0x0000000000b5027b in db::schema_tables::<lambda()>::operator()(void) (__closure=__closure@entry=0x60000086a000) at db/schema_tables.cc:601
#20 0x0000000000b525f2 in apply (args=<optimized out>, func=<optimized out>) at /home/tgrabiec/src/scylla2/seastar/core/apply.hh:34
#21 apply<db::schema_tables::do_merge_schema(distributed<service::storage_proxy>&, std::vector<mutation>, bool)::<lambda()> > (args=<optimized out>, 
    func=<optimized out>) at /home/tgrabiec/src/scylla2/seastar/core/apply.hh:42
#22 do_void_futurize_apply_tuple<db::schema_tables::do_merge_schema(distributed<service::storage_proxy>&, std::vector<mutation>, bool)::<lambda()> > (
    args=<optimized out>, func=<unknown type in /home/tgrabiec/src/scylla2/build/release/scylla, CU 0x9b4600a, DIE 0x9dbd8ac>)
    at /home/tgrabiec/src/scylla2/seastar/core/future.hh:1162
#23 apply<db::schema_tables::do_merge_schema(distributed<service::storage_proxy>&, std::vector<mutation>, bool)::<lambda()> > (args=<optimized out>, 
    func=<optimized out>) at /home/tgrabiec/src/scylla2/seastar/core/future.hh:1182
#24 operator() (__closure=<optimized out>) at /home/tgrabiec/src/scylla2/seastar/core/thread.hh:263
#25 std::_Function_handler<void(), seastar::async(Func&&, Args&& ...)::<lambda(seastar::async(Func&&, Args&& ...)::work&)> mutable [with Func = db::schema_tables::do_merge_schema(distributed<service::storage_proxy>&, std::vector<mutation>, bool)::<lambda()>; Args = {}]::<lambda()> >::_M_invoke(const std::_Any_data &) (__functor=...)
    at /usr/include/c++/5.3.1/functional:1871
#26 0x000000000041627e in operator() (this=0x6000000e1490) at /usr/include/c++/5.3.1/functional:2271
#27 seastar::thread_context::main (this=0x6000000e1480) at core/thread.cc:151
#28 0x0000000000517892 in seastar::thread_context::s_main (lo=<optimized out>, hi=<optimized out>) at core/thread.cc:142
#29 0x00007f7d63497fa0 in ?? () from /lib64/libc.so.6
#30 0x0000000000000000 in ?? ()

```

**Comment by tgrabiec** [comment]
Should be fixed by 0c8db777b1b7f76925437c79dad2115e4d78341e
